src/crypto/asset_proofs.cpp

```
@@ -0,0 +1,952 @@
+    buf.reserve(keys.size() * 32);
+    for (const rct::key* k : keys)
+        buf.insert(buf.end(), k->bytes, k->bytes + 32);
+
+    rct::key result;
+    keccak(buf.data(), (int)buf.size(), result.bytes, 32);
+    sc_reduce32(result.bytes);
+    return result;
+}
+
+// Point addition:  R = A + B  (encoded keys)
+static rct::key point_add(const rct::key& A, const rct::key& B)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)